### ISSUE TITLE
[Feature request] add container memory pressure event

### ISSUE THREAD

**Issue by Colstuwjx** [moby/moby#35817]
As I known, docker will send `kill -9` signal while container OOM and enable its OOMKill option. But it seems that we do NOT support do something while the container is killing due to OOM.

Plans are:
A: send SIGTERM signal to wait container clean exit;
B: register linux eventfd to bind memory pressure and alert to container process.

Both these two plans are tested via [this repo](https://github.com/Colstuwjx/go-oom) and there is no clean way to implement this feature.
Thanks.

**Comment by cpuguy83** [comment]
dockerd does not send any kill command for OOM, this is all handled by the kernel.
When a container is OOM killed, docker fires off an OOM event.

I do not think events for memory pressure are something we should implement here.
dockerd does provide an API to get container stats, including memory usage. This could be implemented using that API.

**Comment by Colstuwjx** [comment]
Yeah, it's a way to succeed, but we need to watch memory usage on each container and do action while met memory pressure.

I think we could add kernel eventfd support, such as watching memory pressure event, and send SIGTERM to container while received that.
